arangod/Replication2/StateMachines/Document/ActiveTransactionsQueue.cpp

```diff
@@ -60,10 +60,14 @@ void ActiveTransactionsQueue::markAsInactive(TransactionId tid) {
   // mark it as inactive. We assume that the log indices are always given in
   // increasing order.
   auto it = _transactions.find(tid);
-  ADB_PROD_ASSERT(it != std::end(_transactions))
-      << "Could not find transaction " << tid;
-  markAsInactive(it->second);
-  _transactions.erase(it);
+  if (it != std::end(_transactions)) {
+    // In some cases (e.g. distribute-shards-like), the leader may be
```

Comment:
I think this is a very good observation. Note that in the `DocumentLeaderState`, the entire data structure is protected, being declared as:
```cpp
Guarded<ActiveTransactionsQueue, std::mutex> _activeTransactions
```
In the `DocumentFollowerState`, it is not protected, but the follower is synchronous so far.
So actually, there's no need to perform any concurrency control inside the `ActiveTransactionsQueue`, because it is already wrapped into it. I realize this is not obvious, so I added an extra comment in https://github.com/arangodb/arangodb/pull/20012/commits/7cc89bbd1be180c658b124ee041fb46f1cb46a4e

Reply:
Thanks